packages/frontend/core/src/modules/peek-view/view/peek-view-controls.tsx

```diff
@@ -140,6 +147,27 @@ export const DocPeekViewControls = ({
           peekView.close(false);
         },
       },
+      {
+        icon: <LinkIcon />,
+        nameKey: 'copy-link',
+        name: t['com.affine.peek-view-controls.copy-link'](),
+        onClick: async () => {
+          const preferredMode = docsService.list.getPrimaryMode(docRef.docId);
+          const search = toURLSearchParams({
```

Comment:
do we have a utility to encode parameters?

Reply:
no we don't. There is one in the server package though. I will add one for the frontend side.